pkg/cluster/cluster.go

```diff
@@ -479,6 +479,15 @@ func (c *Cluster) pollPods() ([]*v1.Pod, []*v1.Pod, error) {
 }
 
 func (c *Cluster) updateStatus() error {
+	if c.bm != nil {
+		bs, err := c.bm.getStatus()
```

Comment:
another idea is to create a go routine to pull backup status every few seconds. so the reconcile loop will not block on it. It complicates the code. I see your points on the unnecessary failure, but how will these transit failures affect the overall user experience?